src/components/Swap/SwapActionButton/SwapButton.tsx

```diff
@@ -90,9 +94,14 @@ export default function SwapButton({ disabled }: { disabled: boolean }) {
       // Only close the review modal if the swap submitted (ie no-throw).
       setOpen(false)
     } catch (e) {
-      console.error(e) // ignore error
+      throwAsync(
+        new SwapError({
+          header: 'Swap Failed',
+          message: swapErrorToUserReadableMessage(e),
```

Comment:
The thrown error has already been converted to a user readable message. You should update those callsites, so that the error here can just be propagated up.

See https://github.com/Uniswap/widgets/blob/main/src/hooks/useUniversalRouter.ts#L58-L66
See https://github.com/Uniswap/widgets/blob/main/src/hooks/swap/useSendSwapTransaction.tsx#L125